### PR TITLE
[BENCH-146] Add numObjects implementation

### PR DESCRIPTION
**Pull request by dexamundsen**
- Add implementation for numObjects in AwsS3StorageFolder
- connected test will be added in BENCH-602 which adds support for Aws Objects (pending design)
- Note: connected test would require putObject permission on AWS (TBD)

manual testing
```
// verily numObjects on CLI - 0
terra-cli (dexamundsen/bench-146) >> terra resource describe --name=dex-s3-519-1
..
# Objects: 0

// write 3 objects into folder on the console (1 file + 1 folder + 1 folder.file)

// verily numObjects on CLI - 2
terra-cli (dexamundsen/bench-146) >> terra resource describe --name=dex-s3-519-1
..
# Objects: 2

// delete objects from folder on the console (1 folder.file, no changes to folder)

// verily numObjects on CLI - 1
terra-cli (dexamundsen/bench-146) >> terra resource describe --name=dex-s3-519-1
..
# Objects: 1

```

### CODE REVIEW

File: src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java [hunk 1]
```diff
@@ -185,6 +193,67 @@ public AwsCredential getControlledAwsS3StorageFolderCredential(
...
+            awsStorageFolder.getRegion());
+
+    try {
+      Iterator<ListObjectsV2Response> listIterator =
+          s3Client
+              .listObjectsV2Paginator(
+                  ListObjectsV2Request.builder()
+                      .bucket(awsStorageFolder.getBucketName())
+                      .prefix(awsStorageFolder.getPrefix() + "/")
+                      .maxKeys(S3_CLIENT_RESULTS_PER_CALL)
```
Comment (jmczerk) [inline src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java:227]: Should `maxKeys` be `Math.min(limit, S3_CLIENT_RESULTS_PER_CALL)`?

**Review by zloery — APPROVED** [review]
LGTM pending John's comment

File: src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java (hunk 1 above)
Comment (jmczerk) [reply src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java:232]: If I remember correctly the `limit` was put in place to make the delete time checks run faster, so I think that's probably the primary consideration here: https://github.com/DataBiosphere/terra-cli/pull/162

Comment (dexamundsen) [reply src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java]: makes sense, done

**Pull request merged**